benchmarks/cosmos-exim/main.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/tendermint/iavl"
+	tmdb "github.com/tendermint/tm-db"
+)
+
+// stores is the list of stores in the CosmosHub database
+// FIXME would be nice to autodetect this
+var stores = []string{
+	"acc",
+	"distribution",
+	"evidence",
+	"god",
+	"main",
+	"mint",
+	"params",
+	"slashing",
+	"staking",
+	"supply",
+	"upgrade",
+}
+
+// Stats track import/export statistics
+type Stats struct {
+	nodes     uint64
+	leafNodes uint64
+	size      uint64
+	duration  time.Duration
+}
+
+func (s *Stats) Add(o Stats) {
+	s.nodes += o.nodes
+	s.leafNodes += o.leafNodes
+	s.size += o.size
+	s.duration += o.duration
+}
+
+func (s *Stats) AddDurationSince(started time.Time) {
+	s.duration += time.Since(started)
+}
+
+func (s *Stats) AddNode(node *iavl.ExportNode) {
+	s.nodes++
+	if node.Height == 0 {
+		s.leafNodes++
+	}
+	s.size += uint64(len(node.Key) + len(node.Value) + 8 + 1)
+}
+
+func (s *Stats) String() string {
+	return fmt.Sprintf("%v nodes (%v leaves) in %v with size %v MB",
+		s.nodes, s.leafNodes, s.duration.Round(time.Millisecond), s.size/1024/1024)
+}
+
+// main runs the main program
+func main() {
+	if len(os.Args) != 2 {
+		fmt.Fprintf(os.Stderr, "Usage: %v <dbpath>\n", os.Args[0])
+		os.Exit(1)
+	}
+	err := run(os.Args[1])
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err.Error())
+		os.Exit(1)
+	}
+}
+
+// run runs the command with normal error handling
+func run(dbPath string) error {
+	version, exports, err := runExport(dbPath)
+	if err != nil {
+		return err
+	}
+
+	err = runImport(version, exports)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// runExport runs an export benchmark and returns a map of store names/export nodes
+func runExport(dbPath string) (int64, map[string][]*iavl.ExportNode, error) {
+	ldb := tmdb.NewDB("application", tmdb.GoLevelDBBackend, dbPath)
+	tree, err := iavl.NewMutableTree(tmdb.NewPrefixDB(ldb, []byte("s/k:main/")), 0)
+	if err != nil {
+		return 0, nil, err
+	}
+	version, err := tree.LoadVersion(0)
+	if err != nil {
+		return 0, nil, err
+	}
+	fmt.Printf("Exporting cosmoshub database at version %v\n\n", version)
+
+	exports := make(map[string][]*iavl.ExportNode, 16)
```

Comment:
why 16?

Reply:
Changed it to the actual number of stores to export instead.